src/chains/algorand_1/models/generalModels.ts

```diff
@@ -1,3 +1,6 @@
+import { EncryptedDataString } from '../../../models'
+import { AlgorandAddress, AlgorandPublicKey, AlgorandPrivateKey } from './cryptoModels'
+
```

Comment:
These comment is for the algoChainState.ts:
ChainEndpoints no longer have a settings property. It was replaced by options - not sure how it re-appeared. Please remove it in this branch.
Algo endpoints should include options.headers = {'X-API-Key': '...'} - which is where the token is specified

Reply:
algoChainState connect() - should call a selectEndpoint function just like Eos plugin. It will return the endpoints[0] for now. The return value of this should be set to _activeEndpoint property of chainState. 
Then, getAlgorandConnectionSettings can use this._activeEndpoint as the URL to parse